### PR TITLE
Fix rkt mounts and add arbitrary volume mounting

### PR DESCRIPTION
**Pull request by schmichael**
Just like #1767 did for the docker driver.

### CODE REVIEW

File: client/driver/rkt.go [hunk 1]
```diff
@@ -42,6 +42,10 @@ const (
 	// The key populated in the Node Attributes to indicate the presence of the
 	// Rkt driver
 	rktDriverAttr = "driver.rkt"
+
+	// rktVolumesConfigOption is the key for enabling the use of custom
+	// bind volumes.
+	rktVolumesConfigOption = "rkt.volumes.enabled"
```
Comment (dadgar) [inline client/driver/rkt.go]: Is this documented on the website? Plus the docker option

File: client/driver/rkt.go [hunk 2]
```diff
@@ -210,15 +215,46 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, e
 		insecure = true
 	}
 	cmdArgs = append(cmdArgs, "run")
-	cmdArgs = append(cmdArgs, fmt.Sprintf("--volume=%s,kind=host,source=%s", task.Name, ctx.AllocDir.SharedDir))
-	cmdArgs = append(cmdArgs, fmt.Sprintf("--mount=volume=%s,target=%s", task.Name, ctx.AllocDir.SharedDir))
+
+	// Mount /alloc
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--volume=%salloc,kind=host,source=%s", task.Name, ctx.AllocDir.SharedDir))
```
Comment (dadgar) [inline client/driver/rkt.go]: I feel like we should be using file path.Join and using the constants

File: client/driver/rkt.go (hunk 1 above)
Comment (schmichael) [inline client/driver/rkt.go]: Added

File: client/driver/rkt.go (hunk 2 above)
Comment (schmichael) [inline client/driver/rkt.go]: To generate the `source=%s` path? Won't that just be reimplementing what's in `ctx.AllocDir.SharedDir`?

Comment (schmichael) [inline client/driver/rkt.go]: Fixed the unique volume names.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ IMPROVEMENTS:
   * client: Introduce a `secrets/` directory to tasks where sensitive data can
     be written [GH-1681]
   * driver: Export `NOMAD_JOB_NAME` environment variable [GH-1804]
+  * driver/rkt: Support rkt volumes (rkt >= 1.0.0 required) [GH-1812]
```
Comment (dadgar) [inline CHANGELOG.md]: Lexicographical ordering, rkt after docker

File: client/driver/rkt.go [hunk 4]
```diff
@@ -210,15 +218,50 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, e
...
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--mount=volume=%s,target=%s", localVolName, allocdir.TaskLocalContainerPath))
+
+	// Mount /secrets
+	secretsVolName := fmt.Sprintf("%s-%s-secrets", ctx.AllocID, task.Name)
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--volume=%s,kind=host,source=%s", secretsVolName, filepath.Join(taskDir, allocdir.TaskSecrets)))
+	cmdArgs = append(cmdArgs, fmt.Sprintf("--mount=volume=%s,target=%s", secretsVolName, allocdir.TaskSecretsContainerPath))
+
+	// Mount arbitrary volumes if enabled
+	if len(driverConfig.Volumes) > 0 {
+		if enabled := d.config.ReadBoolDefault(rktVolumesConfigOption, false); !enabled {
```
Comment (dadgar) [inline client/driver/rkt.go]: Change to default true

File: website/source/docs/drivers/rkt.html.md [hunk 5]
```diff
@@ -75,10 +75,15 @@ The `rkt` driver supports the following configuration in the job spec:
 
 * `debug` - (Optional) Enable rkt command debug option.
 
-## Task Directories
+* `volumes` - (Optional) A list of `host_path:container_path` strings to bind
+  host paths to container paths. Can only be run on clients with the
+  `rkt.volumes.enabled` option set to true.
```
Comment (dadgar) [inline website/source/docs/drivers/rkt.html.md]: The wording should slightly change based on changing the default

File: client/driver/rkt.go [hunk 6]
```diff
@@ -141,7 +149,7 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
 		return false, nil
 	}
 
-	outBytes, err := exec.Command("rkt", "version").Output()
+	outBytes, err := exec.Command(rktCmd, "version").Output()
```
Comment (dadgar) [inline client/driver/rkt.go]: Add rktVolumesConfigOption to the attributes during fingerprinting

Comment (schmichael) [inline client/driver/rkt.go]: Done.

**Pull request merged**